core/src/metadata/metadata_type.rs

```diff
@@ -72,3 +73,64 @@ impl codec::Decode for Metadata {
         subxt_metadata::Metadata::decode(input).map(Metadata::new)
     }
 }
+
+pub trait MetadatExt {
+    fn pallet_by_name_err(
+        &self,
+        name: &str,
+    ) -> Result<subxt_metadata::PalletMetadata, MetadataError>;
+
+    fn pallet_by_index_err(
+        &self,
+        index: u8,
+    ) -> Result<subxt_metadata::PalletMetadata, MetadataError>;
+
+    fn runtime_api_trait_by_name_err(
+        &self,
+        name: &str,
+    ) -> Result<subxt_metadata::RuntimeApiMetadata, MetadataError>;
+
+    fn custom_value_by_name_err(
+        &self,
+        name: &str,
+    ) -> Result<subxt_metadata::CustomValueMetadata, MetadataError>;
+}
+
+impl MetadatExt for subxt_metadata::Metadata {
+    /// Identical to `metadata.pallet_by_name()`, but returns an error if the pallet is not found.
```

Comment:
move the documentation to trait instead :)

Reply:
Still needs doing by the looks of it @tadeohepperle :)